frame/clearing-house/src/lib.rs

```diff
@@ -938,6 +961,101 @@ pub mod pallet {
 			<Self as ClearingHouse>::liquidate(&liquidator_id, &user_id)?;
 			Ok(())
 		}
+
+		/// Set a time for market closure.
+		///
+		/// # Overview
+		///
+		/// If successful, all trading calls to this market are blocked after the timestamp `when`
+		/// passed to this extrinsic. This should allow time for users to close their positions
+		/// normally before the market closes. No one can open positions in this market after the
+		/// extrinsic has been successfully executed, only call
+		/// [`close_position`](Self::close_position) up until the `when` timestamp. After that time,
+		/// all trading calls will fail.
+		///
+		/// Users can settled their positions after the market close by calling
+		/// [`settle_position`](Self::settle_position).
+		///
+		/// TODO(0xangelo): add sequence diagram
+		///
+		/// ## Parameters
+		///
+		/// - `market_id`: the market to be closed
+		/// - `when`: the timestamp at which the market will be closed
+		///
+		/// ## Assumptions or Requirements
+		///
+		/// - Only root (for now) can call this extrinsic
+		/// - The `when` parameter must be a timestamp strictly larger than the current block
```

Comment:
Should there be a minimum window length requirement here? or is strictly larger than the current block enough?

Reply:
That might be configurable, though I'm not sure how yet. Other protocols just close the market immediately after the extrinsic is executed, so I don't have too much reference about how this could be done